src/components/TodoListApp.js

```diff
@@ -0,0 +1,211 @@
+import { v4 as UUID4 } from "uuid";
+
+import InputSection from "./InputSection/InputSection";
+import ListSection from "./Todo/ListSection";
+import StatusSection from "./StatusSection/StatusSection";
+import { FILTER_TYPE } from "../constants/todo.constants";
+
+export default class TodoListApp {
+  constructor($element) {
+    // TodoList 삽입 dom
+    this._$element = $element;
+
+    // 첫 렌더(init) 여부
+    this.isMounted = false;
+
+    // 현재 보기 타입
+    this.filterType = FILTER_TYPE.ALL;
+
+    // 할 일 목록
+    this.todoList = [];
+
+    // instances
+    this.inputSection = null;
+    this.listSection = null;
+    this.statusSection = null;
+
+    // nodes
+    this.$listSection = null;
+    this.$statusSection = null;
+  }
+
+  /**
+   * 입력영역 초기화
+   */
+  resetInput() {
+    this.inputSection.$input.value = "";
+  }
+
+  /**
+   * ONLY jest용 test case 추가를 위해 생성한 메소드
+   * @param {*} type 'ADD' | 'CHECK' | 'UNCHECK'
+   * @param {*} todo
+   * @param {*} targetTodoList
+   */
+  setTodoList(type, todo, targetTodoList) {
+    switch (type) {
+      case "CHECK":
+        targetTodoList.push(todo);
+        break;
+      case "ADD":
+      case "UNCHECK":
+        targetTodoList.unshift(todo);
+        break;
+      default:
+        targetTodoList.unshift(todo);
+        break;
+    }
+  }
+
+  /**
+   * 할 일 추가
+   * @param {string} content 할 일 텍스트
+   * @returns void
+   */
+  addTodo(content) {
+    if (!content.replace(/\s/g, "")) return;
+
+    this.resetInput();
+    this.setTodoList(
+      "ADD",
+      { todoId: UUID4(), isChecked: false, content },
+      this.todoList,
+    );
+
+    this.render();
+
+    return this.todoList;
+  }
+
+  /**
+   * 할 일 체크 / 해제
+   * @param {*} todoId 체크 / 해제 대상 todo의 id
+   */
+  toggleCompleted(todoId) {
+    const targetIndex = this.todoList.findIndex(
+      (todo) => todo.todoId === todoId,
+    );
+    const targetTodo = this.todoList.splice(targetIndex, 1)[0];
+
+    if (targetTodo.isChecked) {
+      targetTodo.isChecked = false;
+      this.setTodoList("UNCHECK", targetTodo, this.todoList);
+    } else {
+      targetTodo.isChecked = true;
+      this.setTodoList("CHECK", targetTodo, this.todoList);
+    }
+
+    this.render();
+  }
+
+  /**
+   * 완료된 할 일 제거
+   */
+  clearCompleted() {
+    const isConfirmed = window.confirm("완료된 할 일을 삭제하시겠습니까?");
+
+    if (isConfirmed) {
+      this.todoList = this.todoList.filter((todo) => !todo.isChecked);
```

Comment:
이렇게 비구조화 할당 할수도 있을것 같아요!
```suggestion
      this.todoList = this.todoList.filter(({isChecked}) => !isChecked);
```

Reply:
명시적인 부분을 좀 좋아하는 취향..? 이라 그렇게 사용하긴 했어용!
cos쪽 작업시 스타일 참고하겠습니다 ㅎㅎ